internal/ingress/controller/parser/parser.go

```diff
@@ -1086,6 +1086,31 @@ func validateProtocol(protocol string) bool {
 	return match
 }
 
+// useSSLProtocol updates the protocol of the route to either https or grpcs, or https and grpcs
+func useSSLProtocol(route *kong.Route) {
+	var http, grpc bool
+	var prots []*string
+
+	for _, val := range route.Protocols {
+
+		if strings.Contains(*val, "grpc") {
+			grpc = true
+		}
+
+		if strings.Contains(*val, "http") {
+			http = true
+		}
+	}
+
+	if grpc {
+		prots = append(prots, kong.String("grpcs"))
+	}
+	if http {
+		prots = append(prots, kong.String("https"))
+	}
+
```

Comment:
This is not correct. Only the secure protocol should be included in `protos`. Else kong will allows insecure (http,grpc) traffic through it as well.

Reply:
I am trying to search if the specified protocol in ingress is http/https or grpc/grpcs. 
I have created a new slice of string **prots** to store the protocols.
If the specified protocol is http/https then I am appending the https to  **prots** and grpcs when grpc/grpcs is specified.
Then assign this slice to route.Protocols.
 Thus ensuring non secure protocol will not be included

Reply:
My bad.
There seems to be an unhandled corner case here. If protocols is not set, then this will default to empty `route.Protocols`, which has a default value of `[ http, https ]` in Kong.
Please set the protocol to `https` when no protocol is present.

Reply:
Ah yes. overrideRouteProtocols method resets prots to empty. I will add code to handle this case.

Reply:
Handled the edge case.